codegen/src/rust.rs

```diff
@@ -523,6 +533,14 @@ pub fn emit_code<W: std::io::Write>(collector: &mut W, ast_node: &AstNode, inden
                 collector.write_all(b";\n\n")?;
             }
             emit_indentation(collector, indentation + 1)?;
+            if parameters.len() < 2
+                || parameters[0].data_type.is_scalar()
+                || parameters[1].data_type.is_scalar()
+                || ["Add", "Sub", "Mul", "Div", "Transformation"].contains(&result.name)
+            {
+                collector.write_all(b"#[inline(always)]\n")?;
+                emit_indentation(collector, indentation + 1)?;
+            }
```

Comment:
This is the least obvious part. The zero and single parameter functions are all simple and should be inlined. The two parameter functions are varying in complexity but the listed ones are always quite simple. All two parameter functions where at least one parameter is a scalar are simple and should be inlined.